### PR TITLE
Update fontface system with scss lists, function and mixin

### PR DESCRIPTION
**Pull request by lucasvallenet**
Update `@font-face` import to reduce code and simplify font management.

**Use :**

`/assets/styles/settings/_config.scss`
```
$font-families: (
    "primary": ("Webfont Primary", "Helvetica Neue", Arial, sans-serif),
    "secondary": ("Webfont Secondary", Georgia, serif),
);

$fontfaces: (
    "Webfont Primary" "webfont-primary_regular" 400 normal,
    "Webfont Primary" "webfont-primary_regular-italic" 400 italic,
    "Webfont Secondary" "webfont-secondary" 700 normal,
);
```

`/assets/styles/**/*.scss`
```
.c-heading.-primary {
    font-family: ff("primary");
}

.c-heading.-secondary {
    font-family: ff("secondary");
}
```

### CODE REVIEW

**Review by mcaskill — CHANGES_REQUESTED** [review]

File: assets/styles/tools/_fonts.scss [hunk 1]
```diff
@@ -2,97 +2,26 @@
...
-        svg:   "#{$file-path}.svg##{$font-family}" format("svg"),
-    );
-
-    @each $key, $values in $formats-map {
-        @if list-contains($file-formats, $key) {
-            $file-path: nth($values, 1);
-            $font-format: nth($values, 2);
-            $src: append($src, url("#{$assets-path}#{$file-path}") $font-format, comma);
+// Mixin
+@mixin fontfaces($webfonts, $dir) {
```
[the remark this answers is not in the document]
Comment (lucasvallenet) [reply assets/styles/tools/_fonts.scss]: Done

File: assets/styles/tools/_fonts.scss [hunk 2]
```diff
@@ -2,97 +2,26 @@
...
-    $file-path,
-    $file-formats: $global-font-file-formats
-) {
-    @font-face {
-        font-family: $font-family;
-        src: font-source-declaration( $font-family, $file-path, $file-formats);
-        @content;
-    }
+// Function
+@function ff($font-family) {
```
[the remark this answers is not in the document]
Comment (lucasvallenet) [reply assets/styles/tools/_fonts.scss]: Done

File: assets/styles/settings/_config.scss [hunk 3]
```diff
@@ -12,14 +12,25 @@ $assets-path: "../"  !default;
...
-$font-sans-serif: sans-serif;
+
+$font-dir: "../fonts/";
+
+$font-families: (
+    "sans": ("Webfont Sans", "Helvetica Neue", Arial, sans-serif),
+    // "serif": ("Webfont Serif", Georgia, serif)
+);
+
+$fontfaces: (
```
[the remark this answers is not in the document]
Comment (lucasvallenet) [reply assets/styles/settings/_config.scss]: Done

File: assets/styles/elements/_fonts.scss [hunk 4]
```diff
@@ -2,35 +2,9 @@
...
-//     "fonts/Bazqux/Regular",
-//     ("eot", "woff2", "woff", "ttf")
-// ) {
-//     font-style:  normal;
-//     font-weight: 400;
-// }
+body {
+    line-height: $line-height;
+    font-family: ff("sans");
+}
```
[the remark this answers is not in the document]
Comment (lucasvallenet) [reply assets/styles/elements/_fonts.scss]: Done

File: assets/styles/elements/_document.scss [hunk 5]
```diff
@@ -5,14 +5,19 @@
...
+
+@include font-faces($font-faces, $font-dir); /* [1] */
+
 html {
     min-height: 100%; /* [2] */
+    line-height: $line-height;
+    font-family: ff("sans");
     color: $color;
-    line-height: $line-height; /* [1] */
+    line-height: $line-height; /* [3] */
```
Comment (mcaskill) [inline assets/styles/elements/_document.scss:20]: Hrm. I never noticed before but the reference numbers are using CSS comments but the referenced texts are using Sass comments.

I suggest we change `/* [N] */` to `// [N]` to not render in the processed (but not minified) CSS file.

**Review by mcaskill — CHANGES_REQUESTED** [review]

File: assets/styles/elements/_document.scss (hunk 5 above)
Comment (lucasvallenet) [reply assets/styles/elements/_document.scss]: Updated

**Review by mcaskill — CHANGES_REQUESTED** [review]

File: assets/styles/objects/_ratio.scss [hunk 6]
```diff
@@ -17,7 +17,7 @@
 
     &:before {
         display: block;
-        padding-bottom: 100%; /* [1] */
+        padding-bottom: 100%; // [1]
```
[the remark this answers is not in the document]
Comment (lucasvallenet) [reply assets/styles/objects/_ratio.scss:20]: Sure it makes sense.
This is not related to this pull request, if validated for merge I can update the comments on other components before than merge ?

[the remark this answers is not in the document]
Comment (lucasvallenet) [reply assets/styles/objects/_ratio.scss]: done

**Pull request merged**